### PR TITLE
Empty states

### PR DESCRIPTION
**Pull request by laulauland**
resolves FP-4114, FP-3810

this introduces some helpful guides in the empty state of some of the live components

### CODE REVIEW

**Review by mies — APPROVED** [review]

File: studio/src/components/KeyboardShortcut.tsx [hunk 1]
```diff
@@ -2,8 +2,8 @@ export const KeyboardShortcutKey = ({
   children,
 }: { children: React.ReactNode }) => {
   return (
-    <span className="flex items-center text-xs justify-center font-mono text-white bg-accent/90 p-0.5 rounded opacity-60 h-4 min-w-4">
+    <kbd className="flex border-[1px] border-gray-300 items-center text-xs justify-center font-mono text-white bg-accent/90 p-0.5 rounded opacity-60 h-4 min-w-4">
```
Comment (flenter) [inline studio/src/components/KeyboardShortcut.tsx]: just adding `border` also adds a 1px border.

File: studio/src/pages/RequestorPage/NavigationPanel/RequestsPanel/RequestsPanel.tsx [hunk 2]
```diff
@@ -156,22 +167,38 @@ export function RequestsPanel() {
...
-            <p className="text-sm text-muted-foreground">No requests found</p>
+    <>
+      <div>
+        <div className="flex items-center space-x-2 pb-3">
+          <div className="relative flex-grow">
+            <Input
+              ref={searchRef}
+              className={cn(
+                "text-sm",
+                isFocused || filterValue ? "pl-2" : "pl-24",
```
Comment (flenter) [inline studio/src/pages/RequestorPage/NavigationPanel/RequestsPanel/RequestsPanel.tsx]: Isn't it possible to use tailwinds selectors for this? There's a `focus:` and also an `empty:` selector.

File: studio/src/pages/RequestorPage/NavigationPanel/RequestsPanel/RequestsPanel.tsx [hunk 3]
```diff
@@ -182,10 +209,61 @@ export function RequestsPanel() {
           />
         ))}
       </div>
-    </div>
+    </>
   );
 }
 
+const EmptyState = memo(function EmptyState() {
```
Comment (flenter) [inline studio/src/pages/RequestorPage/NavigationPanel/RequestsPanel/RequestsPanel.tsx]: Interesting, why is this memoised? This seems to be a fairly straightforward component to render.

File: studio/src/pages/RequestorPage/NavigationPanel/RoutesPanel/RoutesPanel.tsx [hunk 4]
```diff
@@ -11,8 +13,17 @@ import type { ProbedRoute } from "../../types";
...
+  const { routes } = useRequestorStore("routes");
+
+  return (
+    <div className={cn("h-full", "flex", "flex-col")}>
+      <RoutesPanelInner routes={routes} />
+    </div>
+  );
+}
+
+function RoutesPanelInner({ routes }: { routes: ProbedRoute[] }) {
```
Comment (flenter) [inline studio/src/pages/RequestorPage/NavigationPanel/RoutesPanel/RoutesPanel.tsx]: I think i would just merge this with the RoutesPanel component. The parent component is a div + a hook...

File: studio/src/pages/RequestorPage/NavigationPanel/RoutesPanel/RoutesPanel.tsx [hunk 5]
```diff
@@ -163,24 +174,41 @@ export function RoutesPanel() {
...
-            value={filterValue}
-            onChange={(e) => setFilterValue(e.target.value)}
-            onFocus={() => setSelectedRouteIndex(null)}
-          />
+          <div className="relative flex-grow">
+            <Input
+              ref={searchRef}
+              className={cn(
+                "text-sm",
+                isFocused || filterValue ? "pl-2" : "pl-24",
```
Comment (flenter) [inline studio/src/pages/RequestorPage/NavigationPanel/RoutesPanel/RoutesPanel.tsx]: I would love it if this was done using `focus:`   This is the only place relying on  `isFocused` right?

**Review by flenter — APPROVED** [review]
Nice! Left some feedback. I didn't comment at all places (like when memo is used  and situations where there are two components used instead of one slightly larger one)

File: studio/src/pages/RequestorPage/NavigationPanel/RequestsPanel/RequestsPanel.tsx (hunk 3 above)
Comment (laulauland) [reply studio/src/pages/RequestorPage/NavigationPanel/RequestsPanel/RequestsPanel.tsx]: i noticed a hiccup when rendering it (toggling between routes/requests tabs) and was experimenting to see if this helped (then forgot to remove it)

**Pull request merged**